### DISCUSSION TITLE
YOLO v8, TensorRT and Python

### DISCUSSION THREAD

**Discussion by Drexk26** [ultralytics/ultralytics#5360] [Q&A]
Hi!
In my project I am using Nvidia Jetson Orin for object detection/segmentation, and I would like to boost the performance of the neural network exploting TensorRT. I am programming in Python and I was asking myself if there was a set of Python methods or libraries that call the TensorRT framework. Could you please help me in this? If there is a guide that I could read or something else.

Thanks!!

**Comment by glenn-jocher** [comment]
@Drexk26 hello!

Great to hear you're using YOLOv8 for your project on the Nvidia Jetson Orin! 🚀 To leverage TensorRT for performance optimization, YOLOv8 supports exporting models to ONNX format, which can then be converted to TensorRT engines.

While we don't provide direct methods or libraries for TensorRT within the YOLOv8 repo, you can follow these general steps:
1. Export your YOLOv8 model to ONNX using the Export mode.
2. Use the `onnx2trt` tool provided by Nvidia or TensorRT's Python API to convert the ONNX model to a TensorRT engine.

For detailed guidance on exporting models, you can refer to the Export section in our documentation. For TensorRT conversion, Nvidia's official TensorRT documentation will be your best resource.

If you have further questions or need assistance with YOLOv8-specific tasks, feel free to reach out. Happy coding! 😊
